### PR TITLE
Add premultiplied option to composite

### PR DESCRIPTION
**Pull request by Andargor**
Currently, sharp automatically premultiplies during composite. However, this may cause issues if the image below is already premultiplied. This additional `premultiplied` option for composite emulates the VIPS behavior of the `--premultiplied` flag.

If true, premultiplication is not performed by sharp. If false or absent, the current existing behavior is performed.

See #1830 and @jcupitt comments (specifically https://github.com/lovell/sharp/issues/1830#issuecomment-519985352 )

### CODE REVIEW

**Comment by lovell** [comment]
Thanks for the PR. Are you able to add a test for this? Perhaps perform the composite with and without this setting and verify via the `assertMaxColourDistance` test utility?

**Comment by Andargor** [comment]
> Thanks for the PR. Are you able to add a test for this? Perhaps perform the composite with and without this setting and verify via the `assertMaxColourDistance` test utility?

Hopefully there's an example in one of your existing test cases I can base myself on to respect your guidelines, if any?

**Comment by lovell** [comment]
The existing composite "blend" test provides an example of using `assertMaxColourDistance`.

https://github.com/lovell/sharp/blob/3fa91bb4ce2ee561fd33b74666c3870ce24e7d97/test/unit/composite.js#L48-L63

**Comment by Andargor** [comment]
@lovell Hopefully I didn't mess up too badly...

File: lib/composite.js [hunk 1]
```diff
@@ -147,6 +148,14 @@ function composite (images) {
         throw is.invalidParameterError('gravity', 'valid gravity', image.gravity);
       }
     }
+    if (is.defined(image.premultiplied)) {
+      if (is.bool(image.premultiplied)) {
+        composite.premultiplied = image.premultiplied;
+      } else {
+        throw is.invalidParameterError('premultiplied', 'boolean', image.premultiplied);
```
Comment (lovell) [inline lib/composite.js]: Please can you add a unit test to cover the line that throws an Error should `image.premultiplied` be defined but not a boolean.

Comment (Andargor) [reply lib/composite.js]: Hopefully this is consistent with the test for "invalid tile"

**Comment by lovell** [comment]
Fantastic, thank you very much for this, it will be in v0.23.1.

**Pull request merged**